File: python/maths/mt_3424.py
def main() :
    MAX = 100
    for i in range (1, MAX): 
        for j in range (1, i) :
            r = ((i+j)**7-i**7-j**7)%mod
            if r == 0 and i % 7 and j % 7 and j+i%7:
                print(i, j)


if __name__ == '__main__':
    mod = 7**7
    main()

File: cpp/maths/mt_3424.cpp
#include <cstdio>
#include <iostream>
#include <cmath>

#define MAX 180
#define ull unsigned long long int
#define mod 823543

int main() {
    for (int a=1; a<MAX; a++) {
        for (int b=1; b<a; b++) {
            if (a*b*(a+b) % 7) {
                if ((ull)(pow(a+b, 7)-pow(a, 7)-pow(b, 7))%mod == 0) {
                    printf("%d %d\n", a, b);
                }
            }
        }
    }
}